### ISSUE TITLE
[oracle] wait/nowait unparsable

### ISSUE THREAD

**Issue by osarier** [sqlfluff/sqlfluff#8097]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linting the example below leads to a 'Found unparsable section' error.

### Expected Behaviour

The example should be parsed correctly.

### Observed Behaviour

Linting the example below gives the following error:
`Line 4, Position 12: Found unparsable section: 'nowait'`
With wait
`Line 4, Position 12: Found unparsable section: 'wait'`

### How to reproduce

Lint the following example:
```sql
select ex_col
from ex_tab
where id = 'abc'
for update of ex_col nowait;
``` 
and also

```sql
select ex_col
from ex_tab
where id = 'abc'
for update of ex_col wait;
``` 

### Dialect

oracle

### Version

4.2.1
python 3.11.3

### Configuration

[sqlfluff]
dialect = oracle

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by Purva-225** [comment]
Hi! I'd be happy to work on this issue if the current PR is no longer active or if additional changes are needed. Please let me know. Thanks!
